mongorestore/mongorestore_test.go

```diff
@@ -583,3 +583,91 @@ func TestFixHashedIndexes(t *testing.T) {
 		})
 	})
 }
+
+func TestRandom(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+	ctx := context.TODO()
+
+	session, err := testutil.GetBareSession()
+	if err != nil {
+		t.Fatalf("No server available")
+	}
+
+	Convey("Test MongoRestore with {autoIndexId: false} in a non-local database's collection", t, func() {
+		dbName := session.Database("testdata")
+
+		defer dbName.Collection("test_auto_idx").Drop(ctx) // drop the collection to clean up resources
+
+		var args []string
+
+		restore, err := getRestoreWithArgs(args...)
+		So(err, ShouldBeNil)
+
+		restore.TargetDirectory = "testdata/test_auto_idx.bson"
+		result := restore.Restore()
+		So(result.Err, ShouldBeNil)
+
+		// Find the collection
+		filter := bson.D{{"name", "test_auto_idx"}}
+		cursor, err := session.Database("testdata").ListCollections(ctx, filter)
+		if err != nil {
+			panic(err)
+		}
+		defer cursor.Close(ctx)
+		if !cursor.Next(ctx) {
+			panic("no matching collection found")
+		}
+		var collInfo struct {
+			Options bson.M
+		}
+		if err = cursor.Decode(&collInfo); err != nil {
+			panic(err)
+		}
+
+		Convey("{autoIndexId: false} should be flipped to true if server version >= 4.0", func() {
```

Comment:
did you manually test this against a pre-4.0 mongodb instance? At the moment, I don't think we actually run our go integration tests against anything earlier than MongoDB 4.0

Reply:
Discussed offline: I've modified the evergreen file (`common.yml`) so that a 3.6 integration test is run.